pkg/scene/widget_align_system.go

```diff
@@ -0,0 +1,200 @@
+package scene
+
+import (
+	"github.com/reearth/reearth-backend/pkg/id"
+)
+
+// WidgetAlignSystem is the layout structure of any enabled widgets that will be displayed over the scene.
+type WidgetAlignSystem struct {
+	inner WidgetZone
+	outer WidgetZone
+}
+
+// WidgetZone is the structure of each layer (inner and outer) of the align system.
+type WidgetZone struct {
+	left   WidgetSection
+	center WidgetSection
+	right  WidgetSection
+}
+
+// WidgetSection is the structure of each section of the align system.
+type WidgetSection struct {
+	top    WidgetArea
+	middle WidgetArea
+	bottom WidgetArea
+}
+
+// WidgetArea has the widgets and alignment information found in each part area of a section.
+type WidgetArea struct {
+	widgetIds []*id.WidgetID
+	align     string
```

Comment:
It is recommended to define consts

```go
const (
	AlignLeft   = "left"
	AlignCenter = "center"
	AlignRight  = "right"
	ZoneInner   = "inner"
	ZoneOuter   = "outer"
)
```

Reply:
Sorry, I get the general idea. I am guessing I can use these to somehow make sure align is only one of those, but I am not sure how to do that correctly in Golang (in this particular situation). When I googled around a bit I couldn't find anything helpful.
If you could explain a little how you are suggesting to use these consts that would help alot!

Reply:
Go's standard library is very useful as an example of good code in Go. For example, net/http package provides each HTTP method name as a const and uses it in various places, making it clear that it is a special constant compared to writing it directly as a string.

https://pkg.go.dev/net/http#pkg-constants

https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/net/http/method.go;l=10

Reply:
Will start using the standard library as a reference!